### ISSUE TITLE
dependsOn error when creating an eks nodegroup in Python

### ISSUE THREAD

**Issue by davidnhuynh** [pulumi/pulumi-eks#601]
Attempting to create an EKS node group using python and receive an error stating:
"'dependsOn' was passed a value that was not a Resource."

Following the examples provided at:
https://www.pulumi.com/docs/guides/crosswalk/aws/eks/#configuring-your-eks-clusters-worker-nodes-and-node-groups

## Steps to reproduce

Python version:
```
❯ venv/bin/python3 --version
Python 3.9.5
```

Version information for pulumi python packages:
```
pulumi             3.9.1
pulumi-aws         4.14.0
pulumi-eks         0.31.0
pulumi-kubernetes  3.5.1
```
Using the following code snippet based off the examples provided in the docs:
```python
cluster = eks.Cluster('my-cluster',
    eks.ClusterArgs(
        vpc_id = vpc_id,
        public_subnet_ids = [public_subnet_ids],
        private_subnet_ids = [private_subnet_ids],
        skip_default_node_group = True,
        instance_roles = [ role1, role2 ],
    )
)

# Now create multiple node groups, each using a different instance profile for each role.

# First, create a node group for fixed compute.
fixed_node_group = eks.NodeGroup('my-cluster-ng1',
    cluster = cluster.core,
    instance_type = 't2.medium',
    desired_capacity = 2,
    min_size = 1,
    max_size = 3,
    labels = {'ondemand': 'true'},
    instance_profile = instance_profile_1,
    opts = pulumi.ResourceOptions(
        parent = cluster,
        providers = {
            'kubernetes': cluster.provider,
        }
    )
)
```

Expected: 
EKS cluster created and nodegroups created and associated with EKS.
Actual: 
```
Pulumi fails with the following stack trace:
pulumi:pulumi:Stack (eks-test-dev):
      error: Program failed with an unhandled exception:
      error: Traceback (most recent call last):
        File "/Users/dhuynhadxnet.com/Documents/Projects/infrastracture/eks-test/venv/lib/python3.9/site-packages/pulumi/runtime/resource.py", line 519, in do_rpc_call
          return monitor.RegisterResource(req)
        File "/Users/dhuynhadxnet.com/Documents/Projects/infrastracture/eks-test/venv/lib/python3.9/site-packages/grpc/_channel.py", line 946, in __call__
          return _end_unary_response_blocking(state, call, False, None)
        File "/Users/dhuynhadxnet.com/Documents/Projects/infrastracture/eks-test/venv/lib/python3.9/site-packages/grpc/_channel.py", line 849, in _end_unary_response_blocking
          raise _InactiveRpcError(state)
      grpc._channel._InactiveRpcError: <_InactiveRpcError of RPC that terminated with:
      status = StatusCode.UNKNOWN
      details = "'dependsOn' was passed a value that was not a Resource."
      debug_error_string = "{"created":"@1628011197.637119000","description":"Error received from peer ipv4:127.0.0.1:62720","file":"src/core/lib/surface/call.cc","file_line":1070,"grpc_message":"'dependsOn' was passed a value that was not a Resource.","grpc_status":2}"
```

**Comment by justinvp** [comment]
I suspect this is the same issue I noted in https://github.com/pulumi/pulumi-eks/issues/566#issuecomment-849890904:

> However, using the older `eks.NewNodeGroup` yields a new error from inside the TypeScript implementation:
> 
> [...]
> 
> `cluster.Core` includes the `VpcCni` resource, which is marshaled as a resource reference, and is used by the `NodeGroup` component. The problem is we do not have a resource registration on the TypeScript side for the `VpcCni` resource, so when the resource reference is deserialized, instead of rehydrating an instance of `VpcCni` resource, the deserializer falls back to the ID string of the resource, which cannot be used with `dependsOn`. The fix will involve adding a resource registration for `VpcCni`.

This was fixed in #590, but we haven't released a new version that includes the fix yet.

**Comment by davidnhuynh** [comment]
Verified that pulling in the newest unreleased version of the python sdk fixes the issue.
Installed the unreleased version via pip:
`pip install pulumi-eks==0.31.1a1627946757`

**Comment by justinvp** [comment]
@davidnhuynh, thanks for trying out the preview release! We've released `v0.32.0` which includes the fix.

Closing this as fixed by #590
